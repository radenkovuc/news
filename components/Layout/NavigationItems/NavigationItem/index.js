import styled from 'styled-components'
import React from 'react'
import withTranslation from '../../../HOCs/withTranslation'
import {Link} from '../../../../i18n'

const Item = styled.div`
  display: flex;
  border-right: 1px solid #444;
  justify-content: center;
  background: ${(props) => (props.isActive ? '#777777' : '#ffffff')};
  color: ${(props) => (props.isActive ? '#ffffff' : '#000000')};
  cursor: pointer;
  flex: 1;

  @media only screen and (min-width: 600px) {
    flex: none;
  }
`
const Text = styled.span`
  padding: 20px;
  font-size: 20px;
  font-family: 'Nunito Sans', sans-serif;
`

const NavigationItem = (props) => {
  const {text, link, isActive, t} = props

  return (
    <Item isActive={isActive}>
      <Link href={link}>
        <Text>{t(text)}</Text>
      </Link>
    </Item>
  )
}

export default withTranslation(NavigationItem)
