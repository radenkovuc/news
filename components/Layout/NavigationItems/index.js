import {HEADER_MENU_ITEMS as headerRoutes} from '../../../common/consts.json'
import styled from 'styled-components'
import React from 'react'
import NavigationItem from './NavigationItem'
import {withRouter} from 'next/router'

const Wrapper = styled.div`
  display: flex;
  flex: 1;
  align-self: center;
  margin: 0;
`

const NavigationItems = (props) => {
  const navItems = []

  // eslint-disable-next-line no-unused-vars
  for (let route in headerRoutes) {
    // eslint-disable-next-line no-prototype-builtins
    if (headerRoutes.hasOwnProperty(route)) {
      navItems.push(
        <NavigationItem
          key={headerRoutes[route].link}
          link={headerRoutes[route].link}
          text={headerRoutes[route].text}
          isActive={props.router && props.router.route === headerRoutes[route].link}
        />
      )
    }
  }

  return <Wrapper>{navItems}</Wrapper>
}

export default withRouter(NavigationItems)
