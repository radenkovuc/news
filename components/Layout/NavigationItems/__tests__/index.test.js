import React from 'react'
import renderer from 'react-test-renderer'
import NavigationItems from '../index'

describe('<NavigationItems />', () => {
  it('match snapshot', () => {
    const tree = renderer.create(<NavigationItems />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
