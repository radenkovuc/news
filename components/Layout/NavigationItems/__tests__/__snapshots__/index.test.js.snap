// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavigationItems /> match snapshot 1`] = `
<div
  className="Wrapper-sc-3didjk-0 ghaWSH"
>
  <div
    className="Item-sc-14w5znm-0 SlJzJ"
  >
    <span
      className="Text-sc-14w5znm-1 jGVkFb"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      TOP_NEWS
    </span>
  </div>
  <div
    className="Item-sc-14w5znm-0 SlJzJ"
  >
    <span
      className="Text-sc-14w5znm-1 jGVkFb"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      CATEGORIES
    </span>
  </div>
  <div
    className="Item-sc-14w5znm-0 SlJzJ"
  >
    <span
      className="Text-sc-14w5znm-1 jGVkFb"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      SEARCH
    </span>
  </div>
</div>
`;
